### ISSUE TITLE
@ComponentScan for many annotated classes will produce too large method

### ISSUE THREAD

**Issue by ThomasKNR** [InsertKoinIO/koin-annotations#147]
**Describe the bug**
In our case, we have several hundred (around 600) annotated classes in our gradle module. Using `@ComponentScan()` will produce single module with single lambda containing everything found. In our case, this lambda is not compilable for JVM as it is too large.

```
exception: exception: java.lang.RuntimeException: Error generating class file org/koin/ksp/generated/AppModuleGencom_thomaskKt.class (compiled from [/Users/thomask/Documents/KoinComponentScanTest/build/generated/ksp/main/kotlin/org/koin/ksp/generated/AppModuleGencom$thomask.kt]): Method too large: org/koin/ksp/generated/AppModuleGencom_thomaskKt.com_thomask_AppModule$lambda$600 (Lorg/koin/core/module/Module;)Lkotlin/Unit;
```

**To Reproduce**
1. Create 600 `@Single` annotated classes
2. Create app module with `@ComponentScan()` annotation
3. Try to build the project
4. Compilation for JVM fails
[I made a very simple project which demonstrates the problem](https://github.com/ThomasKNR/KoinComponentScan/tree/main)

**Expected behavior**
Produced code is compilable to JVM even for large project with hundreds annotated classes found by single `@ComponentScan`

**Koin project used and used version (please complete the following information):**
 `koin-core version 3.5.6`
 `koin-annotations 1.3.1`
 `koin-ksp-compiler 1.3.1`

**Comment by arnaudgiuliani** [comment]
@ThomasKNR is it all your 600 annotated class in the same file?

**Comment by ThomasKNR** [comment]
No, many files. I expect at least well over hundred separate files, each containing one or more classes annotated with @Single.

**Comment by arnaudgiuliani** [comment]
yeah limit is here, generating on the given module that show the limit

**Comment by arnaudgiuliani** [comment]
can you split your scan per package? Having a module per bunch of package to avoid going over 600 definitions. I'v tested 300 definitions in one module it's ok. Need to test at your size then

**Comment by ThomasKNR** [comment]
I provided app that demostrates, it is strictly 600, 599 works... 
Split scan per package is problem with current implentation of KOIN annotation scanning and our project. We are adding DI to current implementation which was without any DI, result is that DI annotations are scattered all over the place. We have added modules for main packages with most DI, but in the end, we need really generic scan - eg. com.example.app, and this one will trigger whole app search and wil cause the method too large.  We have more specific modules defined (eg com.example.app.di), but it doesn't matter in the end, every annotation is consumed by the most generic one.

**Comment by arnaudgiuliani** [comment]
Ok. The problem is that it will ask to split the module generation into several parts and help reconsolidate. 🤔

**Comment by arnaudgiuliani** [comment]
Found a way to split generation per 500: https://github.com/InsertKoinIO/koin-annotations/pull/220

Will be shipped in 2.0
